drivers/bridge/bridge.go

```diff
@@ -715,27 +729,21 @@ func (d *driver) DeleteNetwork(nid string) error {
 		return err
 	}
 
-	// In case of failures after this point, restore the network isolation rules
-	nwList := d.getNetworks()
-	defer func() {
-		if err != nil {
-			if err := n.isolateNetwork(nwList, true); err != nil {
-				logrus.Warnf("Failed on restoring the inter-network iptables rules on cleanup: %v", err)
-			}
-		}
-	}()
-
-	// Remove inter-network communication rules.
-	err = n.isolateNetwork(nwList, false)
-	if err != nil {
-		return err
-	}
-
 	// We only delete the bridge when it's not the default bridge. This is keep the backward compatible behavior.
 	if !config.DefaultBridge {
 		err = netlink.LinkDel(n.bridge.Link)
+		if err != nil {
+			logrus.Errorf("Failed to delete bridge: %v", err)
+			return err
+		}
 	}
 
+	// clean all relevant iptables rules
+	for _, cleanFunc := range n.iptCleanFuncs {
+		if err = cleanFunc(); err != nil {
```

Comment:
Having this defer resetting the err variable is an issue.
The execution of the deferred function defined at original line 696 will be affected by this.